alicloud/resource_alicloud_ots_table.go

```diff
@@ -0,0 +1,245 @@
+				Optional: true,
+			},
+			"primary_key_4_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"primary_key_4_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"time_to_live": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				//ValidateFunc: validateDiskName,
+			},
+			"max_version": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"read_cap": &schema.Schema{  //read capacity
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"write_cap": &schema.Schema{  //write capacity
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+		},
+	}
+}
+
+func getOtsClient() *tablestore.TableStoreClient{
+	/*
+	Get OTS Client
+	 */
+	endpoint := os.Getenv("OTS_ENDPOINT")
+	instanceName := os.Getenv("OTS_INSTANCE_NAME")
+	accessKeyId := os.Getenv("ALICLOUD_ACCESS_KEY")
+	accessKeySecret := os.Getenv("ALICLOUD_SECRET_KEY")
+	client := tablestore.NewClient(endpoint, instanceName, accessKeyId, accessKeySecret)
+	return client
+}
+
+func getPrimaryKeyType(primaryKeyType string) tablestore.PrimaryKeyType{
+	var keyType tablestore.PrimaryKeyType
+	switch primaryKeyType{
+	case "INTEGER":
+		keyType = tablestore.PrimaryKeyType_INTEGER
+	case "STRING":
+		keyType = tablestore.PrimaryKeyType_STRING
+	case "BINARY":
+		keyType = tablestore.PrimaryKeyType_BINARY
+	}
+	return keyType
+}
+
+func resourceAliyunOtsTableCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("Creating OTS Table...")
+	client := getOtsClient()
+
+	tableMeta := new(tablestore.TableMeta)
+
+	table_name := d.Get("table_name").(string)
+	tableMeta.TableName = table_name
+
+	primaryKeyType := d.Get("primary_key_1_type").(string)
+
+	keyType := getPrimaryKeyType(primaryKeyType)
+
+	tableMeta.AddPrimaryKeyColumn(d.Get("primary_key_1_name").(string), keyType)
+
+	primaryKey2Name := d.Get("primary_key_2_name")
+	primaryKey2Type := getPrimaryKeyType(d.Get("primary_key_2_type").(string))
+	//Whether need to check the value of primaryKey2Type?
+	if primaryKey2Name != nil {
+		tableMeta.AddPrimaryKeyColumn(primaryKey2Name.(string), primaryKey2Type)
+	}
+
+	primaryKey3Name := d.Get("primary_key_3_name")
+	primaryKey3Type := getPrimaryKeyType(d.Get("primary_key_3_type").(string))
+	if primaryKey3Name != nil {
+		tableMeta.AddPrimaryKeyColumn(primaryKey3Name.(string), primaryKey3Type)
+	}
+
+	primaryKey4Name := d.Get("primary_key_4_name")
+	primaryKey4Type := getPrimaryKeyType(d.Get("primary_key_4_type").(string))
+	if primaryKey4Name != nil {
+		tableMeta.AddPrimaryKeyColumn(primaryKey4Name.(string), primaryKey4Type)
+	}
+
+	tableOption := new(tablestore.TableOption)
+	tableOption.TimeToAlive = d.Get("time_to_live").(int)
+	tableOption.MaxVersion = d.Get("max_version").(int)
+	reservedThroughput := new(tablestore.ReservedThroughput)
+	//reservedThroughput.Readcap = d.Get("read_cap").(int)
+	//reservedThroughput.Writecap = d.Get("write_cap").(int)
```

Comment:
Why the line 149 and 150 were annotated?

Reply:
OTS SDK doesn't support those two properties. Will remove the lines.